### PR TITLE
SNOW-186380 Support table name with special character by quotation

### PR DESCRIPTION
**Pull request by sfc-gh-mrui**

### CODE REVIEW

File: src/main/scala/net/snowflake/spark/snowflake/io/StageWriter.scala [hunk 1]
```diff
@@ -331,7 +331,7 @@ private[io] object StageWriter {
     val table = params.table.get
     val tempTable =
       TableName(
-        s"${table.name.replace('"', '_')}_staging_${Math.abs(Random.nextInt()).toString}"
+        s"${table.name.replaceAll("[^a-zA-Z0-9_]", "X")}_staging_${Math.abs(Random.nextInt()).toString}"
```
Comment (sfc-gh-bli) [inline src/main/scala/net/snowflake/spark/snowflake/io/StageWriter.scala:334]:
```suggestion
        s"${table.name.replaceAll("\\W", "X")}_staging_${Math.abs(Random.nextInt()).toString}"
```

Comment (sfc-gh-bli) [reply src/main/scala/net/snowflake/spark/snowflake/io/StageWriter.scala:334]: [^a-zA-Z0-9_] == \w
https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html

Comment (sfc-gh-mrui) [reply src/main/scala/net/snowflake/spark/snowflake/io/StageWriter.scala]: Thanks! Change it.

Comment (sfc-gh-bli) [reply src/main/scala/net/snowflake/spark/snowflake/io/StageWriter.scala]: You can click on `Commit Suggestion` button to commit the suggestion.

**Pull request merged**